crates/worker/src/partition/leadership/leader_state.rs

```diff
@@ -491,6 +491,16 @@ impl LeaderState {
                     )));
                 }
             }
+            Action::ForwardPurgeJournalResponse {
```

Comment:
Instead of having a variant for every possible rpc response, could we have a single 

```
Action::ForwardRpcResponse {
  request_id: RequestId,
  rpcResponse: RpcResponse
}
``` 
?

Reply:
I prefer to have a variant for each operation.

Reply:
Note how the action takes the "invocation client" type, not the rpc (which is the implementation detail).